minion/plugins/base.py

```diff
@@ -141,7 +142,33 @@ def report_progress(self, percentage, description):
     def report_issues(self, issues):
         if issues:
             for issue in issues:
-                issue['Id'] = str(uuid.uuid4())
+                # Generate a unique Id for each issue
+                summary = issue["Summary"] if ("Summary" in issue) else ""
+                cwe_id = issue["Classification"]["cwe_id"] if "Classification" in issue and "cwe_id" in issue["Classification"] else ""
+                url = ""
+                param = ""
+                port = ""
+                if "URLs" in issue and issue["URLs"]:
+                    first_url = issue["URLs"][0]
+                    url = first_url["URL"] if "URL" in first_url else ""
+                    param = first_url["Parameter"] if "Parameter" in first_url else ""
+                if "Ports" in issue and issue["Ports"]:
+                    port = issue["Ports"][0]
+                # Case web issue ( Hash => Summary:CWE_ID:URL:parameter )
+                if param:
+                    id = summary + ":" + cwe_id + ":" + url + ":" + param
+                # Case not a web issue ( Hash => Summary:CWE_ID:URL or Summary:CWE_ID:URL:Port )
+                else:
+                    id = summary + ":" + cwe_id + ":" + url
```

Comment:
April> so there are two problems
[12:24] <April> 1) having a unique UUID for an issue, which the new db model should solve
[12:24] <&yvan> yeah, in practice, we shouldn't be using that - each plugin should have an internal unique identifier for each class of issue reported
[12:24] <April> and 2) having a unique UUID for a type of issue
[12:25] <April> so that minion can map a plugin issue to an issue (possibly) already in the database
[12:25] <&yvan> and ideally those should also have mappings to CWE assoications so that we can perform more in depth analysis

Issues should have:
summary, cwe_id, url, uuid, classifier, and unique issues should simply be id = (uuid + url)